pkg/git/git_test.go

```diff
@@ -147,20 +158,107 @@ func Test_validateGitAuth(t *testing.T) {
 			credsDir, cleanup := createTempDir(t)
 			defer cleanup()
 			if tt.wantSSHdir {
-				os.MkdirAll(credsDir+"/.ssh", 0755)
+				err := os.MkdirAll(filepath.Join(credsDir, ".ssh"), fileMode)
+				if err != nil {
+					t.Errorf("Error creating SSH dir: %v", err)
+				}
 			}
 
 			validateGitAuth(logger, credsDir, tt.url)
-			if tt.logMessage != "" {
-				takeAll := log.TakeAll()
-				if len(takeAll) == 0 {
-					t.Fatal("We didn't receive any logging")
-				}
-				gotmsg := takeAll[0].Message
-				if tt.logMessage != gotmsg {
-					t.Errorf("log message: '%s'\n should be '%s'", tt.logMessage, gotmsg)
+			checkLogMessage(tt.logMessage, log, 0, t)
+		})
+	}
+}
+
+func TestUserHasKnownHostsFile(t *testing.T) {
+	tests := []struct {
+		name               string
+		want               bool
+		wantKnownHostsFile bool
+	}{
+		{
+			name:               "known-hosts-file-exists",
+			want:               true,
+			wantKnownHostsFile: true,
+		},
+		{
+			name:               "known-hosts-file-doesnt-exist",
+			want:               false,
+			wantKnownHostsFile: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			homedir, cleanup := createTempDir(t)
+			defer cleanup()
+			if tt.wantKnownHostsFile {
+				os.MkdirAll(filepath.Join(homedir, ".ssh"), fileMode)
+				knownHostsFile := filepath.Join(homedir, sshKnownHostsUserPath)
+				_, err := os.Create(knownHostsFile)
+				if err != nil {
+					t.Fatalf("Could not create test file %s: %v", knownHostsFile, err)
 				}
 			}
+			got, _ := userHasKnownHostsFile(homedir)
+			if got != tt.want {
+				t.Errorf("User has known hosts file got %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestEnsureHomeEnv(t *testing.T) {
+	tests := []struct {
+		name                 string
+		homeenvSet           bool
+		homeenvEqualsHomedir bool
+	}{
+		{
+			name:                 "Homeenv not set",
+			homeenvSet:           false,
+			homeenvEqualsHomedir: true,
+		},
+		{
+			name:                 "Homeenv same as homedir",
+			homeenvSet:           true,
+			homeenvEqualsHomedir: true,
+		},
+		{
+			name:                 "Homeenv different from homedir",
+			homeenvSet:           true,
+			homeenvEqualsHomedir: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			observer, _ := observer.New(zap.InfoLevel)
+			logger := zap.New(observer).Sugar()
+			homedir, cleanup := createTempDir(t)
+			defer cleanup()
+			var homeenv string
+			if tt.homeenvEqualsHomedir {
+				homeenv = homedir
+			} else {
+				homeenv, cleanup = createTempDir(t)
+				defer cleanup()
+			}
+			if tt.homeenvSet {
+				cleanup := setEnv("HOME", homeenv, t)
+				defer cleanup()
+			}
```

Comment:
nit: not sure if that would be better, but this could be in the "test" case themself (as functions) instead of having a boolean for each.

Reply:
I think I prefer as is but if you feel strongly I'm happy to change.

Reply:
😁 yeah it's a nit 😉